lib/resources/aws/aws_iam_root_user.rb

```diff
@@ -46,6 +46,20 @@ def has_mfa_enabled?
     summary_account['AccountMFAEnabled'] == 1
   end
 
+  def has_virtual_mfa_devices?
+    virtual_mfa_devices.each do |device|
+      if %r{arn:aws:iam::\d{12}:mfa\/root-account-mfa-device} =~
+        device['serial_number']
+        return true
+      end
+    end
+    false
+  end
+
+  def has_hardware_mfa_devices?
+    @__hardware_devices ||= !@client.list_mfa_devices.mfa_devices.empty?
```

Comment:
Alas, this isn't the right logic.  `list_mfa_devices` [will list mfa devices for a particular user](https://docs.aws.amazon.com/sdkforruby/api/Aws/IAM/Client.html#list_mfa_devices-instance_method), or the _currently authenticated user_ - which would be the test harness user, not the root user.  You can't pass the username 'root', either:
```
[cwolfe@lodi inspec-review]$ aws --profile inspec-aws-test-default iam list-mfa-devices --user-name root

An error occurred (NoSuchEntity) when calling the ListMFADevices operation: The user with name root cannot be found.
```

So that's no fun.  However, the CIS PDF tells us that if the account has MFA enabled (which we know from `get_account_summary`, and we have a predicate for it in `has_mfa_enabled?`) BUT has no virtual MFA device (which you have added a predicate for), then it must have a hardware MFA.  So I think this can simply be:

```
   has_mfa_enabled? && !has_virtual_mfa_enabled?
```